e2e/tests/replica_rebuilding.robot

```diff
@@ -36,3 +36,64 @@ Reboot Replica Node While Replica Rebuilding
         Then Wait until replica on replica node rebuilt
         And Check data is intact
     END
+
+Delete Volume Node While Replica Rebuilding
+    Given Set node-down-pod-deletion-policy to do-nothing
+    And Create deployment with rwo volume
+    And Write 2048 MB data to deployment
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Delete replica on volume node to trigger replica rebuilding
+        And During replica rebuilding, delete volume node
+
+        Then Wait for volume attached and unknown
+        And Add deleted node back
+        And Wait for volume attached and healthy
+        And Wait for deployment stable
+        And Check deployment data is intact
+    END
+
+Delete Replica Node While Replica Rebuilding
+    Given Set node-down-pod-deletion-policy to do-nothing
+    And Create deployment with rwo volume
+    And Write 2048 MB data to deployment
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Delete replica on replica node to trigger replica rebuilding
+        And During replica rebuilding, delete replica node
+
+        Then Wait for volume attached and degraded
+        And Add deleted node back
+        And Wait for volume attached and healthy
+        And Check deployment data is intact
+    END
+
+Force Drain Volume Node While Replica Rebuilding
+    And Create deployment with rwo volume
+    And Write 2048 MB data to deployment
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Delete replica on volume node to trigger replica rebuilding
+        And Force Drain volume node
+
+        Then Wait for volume attached and degraded
```

Comment:
After force draining, the workload and IM pods should be drained, and the workload should be failed over to another node, right? 

What you mean is the volume is attached to another node together with the workload, but the volume robustness is degraded, because the replica on the drained node is not back yet.

Reply:
Yes.